### DISCUSSION TITLE
@web3modal/wagmi/react: How to change theme inside «createWeb3Modal» when user changes theme?

### DISCUSSION THREAD

**Discussion by Opty1712** [WalletConnect/walletconnect-monorepo#4231] [Developer Support] [answered]
Hi! In v2 I had component `Web3Modal` which was inside React component, so I could pass changed props there `themeMode={theme}` and it will re-render. In v4 I can't find such ability. Please help

**Comment by glitch-txs** [comment]
Hi you can use the theming hook
https://docs.walletconnect.com/web3modal/react/hooks#useweb3modaltheme

**Reply by Opty1712** [reply, marked as answer]
wow, thank you!
